.github/workflows/rust.yml

```diff
@@ -363,9 +363,24 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: "3.8"
+      - name: Install OpenSSL 1.1.1
```

Comment:
Is there no apt package for this?

Reply:
The images is installed with OpenSSL3.

Reply:
But the installed Python3.8 requires OpenSSL 1.1.1 in particular.

Reply:
We either install OpenSSL 1.1.1 like this PR, or use Python which uses OpenSSL 3, or use an image installed with OpenSSL 1.1.1.

Reply:
Indeed https://github.com/apache/arrow-datafusion/pull/7416 simply continues to use the bullseye rust image, and I have filed an upstream bug report for them to fix the action to not rely on an EOL OpenSSL version.